internal/configuration/validator/authentication.go

```diff
@@ -16,6 +16,46 @@ func validateFileAuthenticationBackend(configuration *schema.FileAuthenticationB
 	if configuration.Path == "" {
 		validator.Push(errors.New("Please provide a `path` for the users database in `authentication_backend`"))
 	}
+
+	if configuration.Algorithm == "" {
+		configuration.Algorithm = schema.DefaultFileAuthenticationBackendConfiguration.Algorithm
+	} else {
+		configuration.Algorithm = strings.ToLower(configuration.Algorithm)
+		if configuration.Algorithm != "argon2id" && configuration.Algorithm != "sha512" {
+			validator.Push(fmt.Errorf("Unknown hashing algorithm supplied, valid values are argon2id and sha512, you provided '%s'", configuration.Algorithm))
+		}
+	}
+
+	if configuration.Rounds == 0 {
+		if configuration.Algorithm == "argon2id" {
+			configuration.Rounds = schema.DefaultFileAuthenticationBackendConfiguration.Rounds
+		} else {
+			configuration.Rounds = schema.DefaultFileAuthenticationBackendSHA512Configuration.Rounds
+		}
+	} else if configuration.Rounds < 0 {
+		validator.Push(fmt.Errorf("The number of rounds specified is invalid, must be more than 0 but you specified %d", configuration.Rounds))
```

Comment:
```suggestion
		validator.Push(fmt.Errorf("The number of rounds specified is invalid, must be more than 0, you configured %d", configuration.Rounds))
```

Reply:
Done like this for all instances. Also is now iterations.